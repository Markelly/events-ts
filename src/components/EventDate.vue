<template>
  <div class="date">{{ date }}</div>
</template>

<script lang="ts">
import { monthName, weekDayName } from "@/utils/constants";
import { Component, Prop, Vue } from "vue-property-decorator";
import { DateType } from "@/store/data-types";

@Component
export default class EventDate extends Vue {
  @Prop()
    dates!: DateType;

    get date(): string {
      let formattedDate: string = "";
      const date = new Date(this.dates?.start?.localDate);
      if (date) {
        let weekday = weekDayName[date.getDay()];
        let month = monthName[date.getMonth()];
        let day = date.getDate();
        let year = date.getFullYear();
        formattedDate = `${weekday}, ${day} ${month} ${year}`;
      }
      return formattedDate;
    }
};
</script>

<style scoped lang="scss">
@import "@/styles/events-ts";

.date {
  color: $color-primary;
  font-weight: 600;
  padding: 4px;
  font-size: 12px;

  @media (min-width: $breakpoint-tablet) {
    font-size: 16px;
    padding: 8px;
  }
}
</style>
