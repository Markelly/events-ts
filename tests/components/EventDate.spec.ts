import EventDate from '@/components/EventDate.vue';
import { createLocalVue, shallowMount } from '@vue/test-utils';

const localVue = createLocalVue();

describe('<EventDate/>', () => {
  let wrapper: any;

  beforeEach(() => {
    wrapper = shallowMount(EventDate, {
      localVue,
      propsData: {
        EventDate: {
          start: {
            localDate: '2020-04-03'
          }
        }
      }
    });
  });

  test('should render the component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
