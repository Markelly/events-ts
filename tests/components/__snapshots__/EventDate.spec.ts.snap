// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EventDate/> should render the component 1`] = `
VueWrapper {
  "_emitted": Object {},
  "_emittedByOrder": Array [],
  "isFunctionalComponent": undefined,
}
`;
